File: app/src/main/java/com/example/gotapp/PersonajesActivity.java
package com.example.gotapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class PersonajesActivity extends AppCompatActivity {

    private RecyclerView rvPersonajes;
    PersonajeAdapter adapter;
    private Toolbar mainToolbar;

    @Override
    public boolean onCreateOptionsMenu(Menu menu){
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return super.onCreateOptionsMenu(menu);
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == R.id.volver) {
            Intent intent = new Intent(PersonajesActivity.this, HomePage.class);
            startActivity(intent);
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_personajes);
        setUpAdapter();
        mainToolbar = findViewById(R.id.main_toolbar);
        setSupportActionBar(mainToolbar);
        getSupportActionBar().setTitle("Game of Thrones");
    }

    private void setUpAdapter() {
        rvPersonajes = findViewById(R.id.rv_personajes);
        adapter = new PersonajeAdapter(getPersonajes());
        rvPersonajes.setAdapter(adapter);
    }

    private List<Personaje> getPersonajes() {
        return new ArrayList<Personaje>(){{
                add(new Personaje(1, null, "Jhon", "Masculino", "10/5/1990", "yankee"));
                add(new Personaje(2, null, "Smith", "Masculino", "10/5/1991", "yankee"));
                add(new Personaje(3, null, "Simon", "Masculino", "10/5/1992", "yankee"));
                add(new Personaje(4, null, "Jessica", "Femenino", "10/5/1996", "yankee"));
                add(new Personaje(5, null, "Emily", "Femenino", "10/5/1994", "yankee"));
            }};
    }
}

File: app/src/main/java/com/example/gotapp/PersonajesActivity.kt
package com.example.gotapp

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.gotapp.databinding.ActivityPersonajesBinding
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import retrofit2.Response
import java.util.ArrayList

class PersonajesActivity : AppCompatActivity() {
    private lateinit var binding: ActivityPersonajesBinding
    private lateinit var rvPersonajes: RecyclerView
    var adapter: PersonajeAdapter? = null
    private var mainToolbar: Toolbar? = null

    private val personajesImagenes = mutableListOf<String>()

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.volver) {
            val intent = Intent(this@PersonajesActivity, HomePage::class.java)
            startActivity(intent)
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_personajes)
        setUpAdapter()
        mainToolbar = findViewById(R.id.main_toolbar)
        setSupportActionBar(mainToolbar)
        supportActionBar!!.title = "Game of Thrones"
        initRecyclerView()
    }

    private fun initRecyclerView(){
        adapter = PersonajeAdapter(personajesImagenes)
        binding.rvPersonajes.layoutManager = LinearLayoutManager(this)
        binding.rvPersonajes.adapter = adapter


    }

  /*  private fun setUpAdapter() {
        rvPersonajes = findViewById(R.id.rv_personajes)
        adapter = PersonajeAdapter(personajes)
        rvPersonajes.setAdapter(adapter)
    }*/


    private fun personajesGet(){
        CoroutineScope(Dispatchers.IO).launch {
            val call: Response<List<PersonajesResponse>> = RetrofitClient.retrofit.create(GotApi::class.java).getCharacters()
            val characters: List<PersonajesResponse>? =  call.body()
            var aux: PersonajesResponse ?=null
            if (characters != null) {
                for(s in characters){
                    aux = s
                }
            }
            runOnUiThread{
                if(call.isSuccessful){
                    //TODO
                    Log.i("Characters", characters.toString())
                    val images : String? = aux?.url


                }else{
                    Log.i("Error","Error")

                }
            }



        }

    }



    private val personajes: ArrayList<Personaje?>
        get() {
            TODO()
        }
}